projects/story-decorator/decorator.ts

```diff
@@ -78,6 +78,7 @@ export const swcThemeDecoratorWithConfig =
             if (context.globals.textDirection) {
                 window.__swc_hack_knobs__.defaultDirection =
                     context.globals.textDirection;
+                document.documentElement.dir = context.globals.textDirection;
```

Comment:
Instead of assigning `document.documentElement.dir` every time, check if it actually needs to be updated. This will prevent unnecessary DOM mutations.
```js
if (context.globals.textDirection && document.documentElement.dir !== context.globals.textDirection) {
    document.documentElement.dir = context.globals.textDirection;
}
```